source/SIL.AppBuilder.Portal.Frontend/src/ui/components/product-artifact/artifacts.tsx

```diff
@@ -1,111 +1,70 @@
 
-  render() {
-    const { artifacts, t } = this.props;
-    const { areArtifactsVisible } = this.state;
+  const productDefinition = dataStore.cache.query((q) =>
+    q.findRelatedRecord(product, 'productDefinition')
+  );
+  let artifacts = [];
 
-    return (
-      <div data-test-product-artifacts className='product-artifact w-100 m-b-lg'>
-        <Header
-          {...this.props}
-          onClick={this.toggleShowArtifacts}
-          isCollapsed={areArtifactsVisible}
-        />
+  if (productBuild) {
+    artifacts = dataStore.cache.query((q) =>
+      q.findRelatedRecords(productBuild, 'productArtifacts')
+    );
+  }
 
-        {areArtifactsVisible && (
-          <div data-test-artifact-list-container>
-            <EmptyLabel
-              className='m-t-lg m-l-lg m-b-lg'
-              condition={!isEmpty(artifacts)}
-              label={t('project.products.noArtifacts')}
-            >
-              <div className='flex p-l-md p-t-sm p-b-sm p-r-md gray-text bold'>
-                <div className='flex-70'>
-                  <span>{t('project.products.filename')}</span>
+  return (
+    <div data-test-product-artifacts className='product-artifact w-100 m-b-lg'>
+      <Header
+        {...{
+          productDefinition,
+          artifacts,
+          product,
+        }}
+        onClick={toggleShowArtifacts}
+        isCollapsed={areArtifactsVisible}
+      />
+
+      {areArtifactsVisible && (
+        <div data-test-artifact-list-container>
+          <EmptyLabel
+            className='m-t-lg m-l-lg m-b-lg'
+            condition={!isEmpty(artifacts)}
+            label={t('project.products.noArtifacts')}
+          >
+            <div className='flex p-l-md p-t-sm p-b-sm p-r-md gray-text bold'>
+              <div className='flex-70'>
+                <span>{t('project.products.filename')}</span>
+              </div>
+              <div className='flex flex-30'>
+                <div className='flex-grow'>
+                  <span>{t('project.products.updated')}</span>
                 </div>
-                <div className='flex flex-30'>
-                  <div className='flex-grow'>
-                    <span>{t('project.products.updated')}</span>
-                  </div>
-                  <div className='flex-30 text-align-right'>
-                    <span className='m-r-md'>{t('project.products.size')}</span>
-                  </div>
-                  <div className='flex-10' />
+                <div className='flex-30 text-align-right'>
+                  <span className='m-r-md'>{t('project.products.size')}</span>
                 </div>
+                <div className='flex-10' />
               </div>
-
-              {(artifacts || []).map((artifact) => (
-                <Artifact key={artifact.id} artifact={artifact} />
-              ))}
-            </EmptyLabel>
-          </div>
-        )}
-      </div>
-    );
-  }
+            </div>
+
+            {(artifacts || []).map((artifact) => (
```

Comment:
why doesn't findRelatedRecords just return an empty array? feels like that would remove the need for this type of checking throughout the app?

Reply:
that's a good thing to investigate. I'll have to check on this.